src/parser.ts

```diff
@@ -13,6 +13,21 @@ let PlaceHolders = {
     ArrowParameterPlaceHolder: 'ArrowParameterPlaceHolder'
 };
 
+function upgradeToken(token) {
+    if (token && state.strict && token.type === Token.Identifier) {
```

Comment:
Even here, you'd be able to avoid the double nested `if` if we had a function that combined `state.strict && scanner.isStrictModeReservedWord`